chess/ChessGameModel.swift

```diff
@@ -174,8 +190,44 @@ extension ChessGameModel {
 
 }
 
+// MARK: - Helpers -
+
 extension ChessGameModel {
 
+    enum MessageKind {
+        case move
+        case color
+        case checkmate
+        case invalidMove
+
+        init?(_ message: String) {
+            if Self.isValidNotation(message) {
```

Comment:
next time instead of these functions I would recommend to write regex statements for each of these cases. Then the init will look much simpler